web-ng/src/app/services/data-store/data-store.service.ts

```diff
@@ -55,6 +55,32 @@ export class DataStoreService {
       );
   }
 
+  /**
+   * Returns an Observable that loads and emits the projects to which the user
+   * is part of.
+   *
+   */
+  loadAccessibleProject$(userEmail: string): Observable<Project[]> {
+    return this.db
+      .collection('projects')
+      .snapshotChanges()
+      .pipe(
+        map(actions =>
+          actions
+            .filter(a => {
+              const data = a.payload.doc.data() as Project;
+              const emails = Object.keys(data.acl);
+              return emails.indexOf(userEmail) > -1;
+            })
+            .map(a => {
+              const data = a.payload.doc.data() as Project;
```

Comment:
data() isn't of type Project, it's of type DocumentData, so it shouldn't be cast here and above. We use `FirebaseDataConverter.toProject()` to create a valid Project instance below.

Reply:
updated.